src/FakeItEasy/Core/FakeManagerProvider.cs

```diff
@@ -81,13 +81,19 @@ public void EnsureInitialized(object proxy)
                 {
                     this.initializedFakeManager = this.fakeManagerFactory(this.typeOfFake, proxy);
 
-                    this.fakeManagerAccessor.TagProxy(proxy, this.initializedFakeManager);
+                    this.fakeManagerAccessor.SetFakeManager(proxy, this.initializedFakeManager);
 
                     this.ApplyInitialConfiguration(proxy);
                 }
             }
         }
 
+        public void EnsureManagerIsRegistered()
+        {
+            var accessor = ServiceLocator.Current.Resolve<IFakeManagerAccessor>();

```

Comment:
How 'bout we just make `this.fakeManagerAccessor` serializable? Everything works except for `FakeManagerProviderTests.Should_be_able_to_fetch_a_serialized_and_deserialized_initialized_provider`, but I think that's because of all the fakes _it's_ made of. And I think the test is redundant.